### PR TITLE
[exporter/datadog] Deprecate `config.Config.OnceMetadata` method

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

Deprecate `OnceMetadata` method from Config struct. This method was only here because the configuration is in a different package.

External users can't really leverage this for anything useful, but I am trying to follow the guidelines and deprecate it in two steps.

**Link to tracking Issue:** #8373

**Testing:** 

- [x] Amended unit tests.
- [x] End to end test when sending metrics and traces with `only_metadata` set to `true` and to `false`

### CODE REVIEW

**Review by KSerrania — APPROVED** [review]

**Review by jpkrohling — COMMENTED** [review]
Approving by proxy, as the author is a code owner and has received approval from another code owner.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
